### ISSUE TITLE
Missing noexcept specifiers

### ISSUE THREAD

**Issue by InonS** [ARM-software/ComputeLibrary#447]
**Problem description:**
Compilation fails due to multiple `noexcept` specifier inconsistencies. 
`scons -j 4 arch=x86_64  extra_cxx_flags="-fPIC" benchmark_tests=1 validation_tests=1 opencl=1 embed_kernels=1`

Removing the compilation flag allows the build to complete.
```
diff --git a/SConstruct b/SConstruct
index 532001a7..7e2b0827 100644
--- a/SConstruct
+++ b/SConstruct
@@ -104,7 +104,7 @@ if env['os'] == 'android' and ( 'clang++' not in cpp_compiler or 'clang' not in
 if 'clang++' in cpp_compiler:
     env.Append(CXXFLAGS = ['-Wno-format-nonliteral','-Wno-deprecated-increment-bool','-Wno-vla-extension','-Wno-mismatched-tags'])
 else:
-    env.Append(CXXFLAGS = ['-Wlogical-op','-Wnoexcept','-Wstrict-null-sentinel'])
+    env.Append(CXXFLAGS = ['-Wlogical-op','-Wstrict-null-sentinel']) # -Wnoexcept
```

**Output of 'strings libarm_compute.so | grep arm_compute_version':**
`arm_compute_version=v18.05 Build options: {'embed_kernels': '1', 'opencl': '1', 'validation_tests': '1', 'benchmark_tests': '1', 'extra_cxx_flags': '-fPIC', 'arch': 'x86_64'} Git hash=b3a371bc429d2ba45e56baaf239d8200c2662a74N11arm_compute9IFunctionE`

**Platform:**
`x86_64`

**Operating System:**
`Linux 4.16.10-1-ARCH #1 SMP PREEMPT x86_64 GNU/Linux`

**Comment by AnthonyARM** [comment]
We haven't had this issue, which version of the compiler are you using?

**Comment by InonS** [comment]
```
SCons by Steven Knight et al.:
	script: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog
	engine: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog
	engine path: ['/usr/lib/python2.7/site-packages/SCons']
```

`gcc 8.1.0`

**Comment by mdigiorgio** [comment]
Now fixed in v18.08. Thanks for reporting.
